contracts/reward/StableYield.sol

```diff
@@ -0,0 +1,176 @@
+            "Wrong input parameters"
+        );
+        require(
+            (STABLE_YIELD_BASE * totalSupply * totalSupply) /
+                totalSupply /
+                STABLE_YIELD_BASE ==
+                totalSupply,
+            "Potential overflow"
+        );
+        token = _token;
+        tokenStaking = _tokenStaking;
+        _transferOwnership(_msgSender());
+    }
+
+    /// @notice Sets or updates application parameter for minting reward.
+    ///         Can be called only by the governance.
+    function setApplicationParameters(
+        address application,
+        uint256 stableYield,
+        uint256 duration,
+        address distributor,
+        string memory receiveRewardMethod
+    ) external onlyOwner {
+        // if stable yield is zero then reward will be no longer minted
+        require(
+            (stableYield == 0 ||
+                (stableYield < STABLE_YIELD_BASE && duration > 0)) &&
+                distributor != address(0),
+            "Wrong input parameters"
+        );
+        ApplicationInfo storage info = applicationInfo[application];
+        info.stableYield = stableYield;
+        info.duration = duration;
+        info.distributor = distributor;
+        info.receiveRewardMethod = receiveRewardMethod;
+        emit ParametersSet(
+            application,
+            stableYield,
+            duration,
+            distributor,
+            receiveRewardMethod
+        );
+    }
+
+    /// @notice Mints reward and then pushes it to particular application or distributor.
+    /// @dev Application must be in `APPROVED` state
+    function mintAndPushReward(address application) external {
+        ApplicationInfo storage info = applicationInfo[application];
+        require(
+            info.stableYield != 0,
+            "Reward parameters are not set for the application"
+        );
+        require(
+            /* solhint-disable-next-line not-rely-on-time */
+            block.timestamp >= info.lastMint + info.duration,
+            "New portion of reward is not ready"
+        );
+        IStaking.ApplicationStatus status = tokenStaking.getApplicationStatus(
+            application
+        );
+        require(
+            status == IStaking.ApplicationStatus.APPROVED,
+            "Application is not approved"
+        );
+        uint96 reward = caclulateReward(application, info.stableYield);
+        /* solhint-disable-next-line not-rely-on-time */
+        info.lastMint = block.timestamp;
+        //slither-disable-next-line incorrect-equality
+        if (bytes(info.receiveRewardMethod).length == 0) {
+            sendToDistributor(info.distributor, reward);
+        } else {
+            executeReceiveReward(application, info.receiveRewardMethod, reward);
+        }
+        emit MintedReward(application, reward);
+    }
+
+    function sendToDistributor(address distributor, uint96 reward) internal {
+        token.mint(distributor, reward);
```

Comment:
Given the criticality of minting tokens, what do you think if we add a manual approval step (e.g., from the council multisig)?

Reply:
I was hoping to have `mintAndPush` as permissionless method. Can you elaborate how this will work with council? in which moment? Like sort of queue?